src/librbd/crypto/CryptoObjectDispatch.cc

```diff
@@ -324,10 +326,40 @@ struct C_UnalignedObjectWriteRequest : public Context {
       }
     }
 
+    bool trigger_copyup() {
+      auto ctx = create_context_callback<
+              C_UnalignedObjectWriteRequest<I>,
+              &C_UnalignedObjectWriteRequest<I>::handle_copyup>(this);
+      bufferlist bl;
+      auto req = new io::ObjectWriteRequest<I>(image_ctx, object_no, 0,
```

Comment:
I would instead use the `ObjectDispatchSpec::create_write` to keep things consistent -- just provided `CRYPTO` as the layer enum so it starts below this layer.

Reply:
What if the object/image has no parent?
In that case the write will turn to be a wasted op, no?

Using the core layer directly allows me to query `has_parent()`.
Furthermore, since I'm expecting to get processed by core layer, don't you think it's better to avoid going through other layers (which in the future perhaps can discard this type of write)?

Reply:
I think it might be cleaner to just have a `librbd::io::util` function to do that math for you.

Reply:
... or perhaps move that common logic from FlattenRequest and here to a new helper function in librbd::io::util